### PR TITLE
Add zone records endpoints support

### PR DESCRIPTION
**Pull request by jacegu**
Adds support for: https://developer.dnsimple.com/v2/zones/records/

### CODE REVIEW

File: lib/dnsimple/zones_service.ex [hunk 1]
```diff
@@ -28,4 +28,69 @@ defmodule Dnsimple.ZonesService do
...
+  @doc """
+  Updates a record in the zone.
+
+  See: https://developer.dnsimple.com/v2/zones/records/#update
+  """
+  @spec update_record(Client.t, String.t, String.t, Integer.t, Keyword.t, Keyword.t, Keyword.t) :: Response.t
+  def update_record(client, account_id, zone_name, record_id, attributes, headers \\ [], options \\ []) do
+    url = Client.versioned("/#{account_id}/zones/#{zone_name}/records/#{record_id}")
+
+    Client.put(client, url, attributes, headers, options)
```
Comment (weppos) [inline lib/dnsimple/zones_service.ex]: The updates should be `PATCH`, not `PUT`.

Comment (jacegu) [inline lib/dnsimple/zones_service.ex]: True. Sorry about that.

**Comment by jacegu** [comment]
List records working:

```
iex(2)> Dnsimple.ZonesService.list_records(client, 63, "aaa-0001.com")
{:ok,
 %Dnsimple.Response{data: [%Dnsimple.Record{content: "ns1.dnsimple.com admin.dnsimple.com 1457638723 86400 7200 604800 300",
    created_at: "2016-03-10T19:38:41.902Z", id: 66945, name: "", parent_id: nil,
    priority: nil, system_record: true, ttl: 3600, type: "SOA",
    updated_at: "2016-05-05T11:41:26.211Z", zone_id: "aaa-0001.com"},
   %Dnsimple.Record{content: "ns1.dnsimple.com",
    created_at: "2016-03-10T19:38:41.962Z", id: 66946, name: "", parent_id: nil,
    priority: nil, system_record: true, ttl: 3600, type: "NS",
    updated_at: "2016-03-10T19:38:41.962Z", zone_id: "aaa-0001.com"},
   %Dnsimple.Record{content: "ns2.dnsimple.com",
    created_at: "2016-03-10T19:38:41.976Z", id: 66947, name: "", parent_id: nil,
    priority: nil, system_record: true, ttl: 3600, type: "NS",
    updated_at: "2016-03-10T19:38:41.976Z", zone_id: "aaa-0001.com"},
   %Dnsimple.Record{content: "ns3.dnsimple.com",
    created_at: "2016-03-10T19:38:41.990Z", id: 66948, name: "", parent_id: nil,
    priority: nil, system_record: true, ttl: 3600, type: "NS",
    updated_at: "2016-03-10T19:38:41.990Z", zone_id: "aaa-0001.com"},
   %Dnsimple.Record{content: "ns4.dnsimple.com",
    created_at: "2016-03-10T19:38:42.004Z", id: 66949, name: "", parent_id: nil,
    priority: nil, system_record: true, ttl: 3600, type: "NS",
    updated_at: "2016-03-10T19:38:42.004Z", zone_id: "aaa-0001.com"},
   %Dnsimple.Record{content: "127.0.0.1",
    created_at: "2016-05-05T11:37:41.769Z", id: 148947, name: "www",
    parent_id: nil, priority: nil, system_record: false, ttl: 3600, type: "A",
    updated_at: "2016-05-05T11:41:26.200Z", zone_id: "aaa-0001.com"}],
  http_response: %HTTPoison.Response{body: "{\"data\":[{\"id\":66945,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"\",\"content\":\"ns1.dnsimple.com admin.dnsimple.com 1457638723 86400 7200 604800 300\",\"ttl\":3600,\"priority\":null,\"type\":\"SOA\",\"system_record\":true,\"created_at\":\"2016-03-10T19:38:41.902Z\",\"updated_at\":\"2016-05-05T11:41:26.211Z\"},{\"id\":66946,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"\",\"content\":\"ns1.dnsimple.com\",\"ttl\":3600,\"priority\":null,\"type\":\"NS\",\"system_record\":true,\"created_at\":\"2016-03-10T19:38:41.962Z\",\"updated_at\":\"2016-03-10T19:38:41.962Z\"},{\"id\":66947,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"\",\"content\":\"ns2.dnsimple.com\",\"ttl\":3600,\"priority\":null,\"type\":\"NS\",\"system_record\":true,\"created_at\":\"2016-03-10T19:38:41.976Z\",\"updated_at\":\"2016-03-10T19:38:41.976Z\"},{\"id\":66948,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"\",\"content\":\"ns3.dnsimple.com\",\"ttl\":3600,\"priority\":null,\"type\":\"NS\",\"system_record\":true,\"created_at\":\"2016-03-10T19:38:41.990Z\",\"updated_at\":\"2016-03-10T19:38:41.990Z\"},{\"id\":66949,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"\",\"content\":\"ns4.dnsimple.com\",\"ttl\":3600,\"priority\":null,\"type\":\"NS\",\"system_record\":true,\"created_at\":\"2016-03-10T19:38:42.004Z\",\"updated_at\":\"2016-03-10T19:38:42.004Z\"},{\"id\":148947,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"www\",\"content\":\"127.0.0.1\",\"ttl\":3600,\"priority\":null,\"type\":\"A\",\"system_record\":false,\"created_at\":\"2016-05-05T11:37:41.769Z\",\"updated_at\":\"2016-05-05T11:41:26.200Z\"}],\"pagination\":{\"current_page\":1,\"per_page\":30,\"total_entries\":6,\"total_pages\":1}}",
   headers: [{"Server", "nginx"}, {"Date", "Thu, 05 May 2016 16:29:54 GMT"},
    {"Content-Type", "application/json; charset=utf-8"},
    {"Transfer-Encoding", "chunked"}, {"Connection", "keep-alive"},
    {"Status", "200 OK"}, {"X-RateLimit-Limit", "2400"},
    {"X-RateLimit-Remaining", "2399"}, {"X-RateLimit-Reset", "1462469393"},
    {"ETag", "W/\"f1b27dc0e0e98a05b98f39464b42cd98\""},
    {"Cache-Control", "max-age=0, private, must-revalidate"},
    {"X-Request-Id", "7ab75b78-8d81-458f-8371-88e45abfcd91"},
    {"X-Runtime", "0.144038"},
    {"Strict-Transport-Security", "max-age=31536000"}], status_code: 200},
  rate_limit: 2400, rate_limit_remaining: 2399, rate_limit_reset: 1462469393}}
```

**Comment by jacegu** [comment]
Create record working:

```
iex(4)> Dnsimple.ZonesService.create_record(client, 63, "aaa-0001.com", %{name: "www", type: "mx", content: "mxa.example.com", priority: 10, ttl: 600})
{:ok,
 %Dnsimple.Response{data: %Dnsimple.Record{content: "mxa.example.com",
   created_at: "2016-05-05T16:33:55.810Z", id: 149267, name: "www",
   parent_id: nil, priority: 10, system_record: false, ttl: 600, type: "MX",
   updated_at: "2016-05-05T16:33:55.810Z", zone_id: "aaa-0001.com"},
  http_response: %HTTPoison.Response{body: "{\"data\":{\"id\":149267,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"www\",\"content\":\"mxa.example.com\",\"ttl\":600,\"priority\":10,\"type\":\"MX\",\"system_record\":false,\"created_at\":\"2016-05-05T16:33:55.810Z\",\"updated_at\":\"2016-05-05T16:33:55.810Z\"}}",
   headers: [{"Server", "nginx"}, {"Date", "Thu, 05 May 2016 16:33:55 GMT"},
    {"Content-Type", "application/json; charset=utf-8"},
    {"Transfer-Encoding", "chunked"}, {"Connection", "keep-alive"},
    {"Status", "201 Created"}, {"X-RateLimit-Limit", "2400"},
    {"X-RateLimit-Remaining", "2398"}, {"X-RateLimit-Reset", "1462469393"},
    {"ETag", "W/\"1cd90f69f4e3dc9afeb5f3aa134c1489\""},
    {"Cache-Control", "max-age=0, private, must-revalidate"},
    {"X-Request-Id", "40bb321e-e795-462b-92fb-1534740148ea"},
    {"X-Runtime", "0.160589"},
    {"Strict-Transport-Security", "max-age=31536000"}], status_code: 201},
  rate_limit: 2400, rate_limit_remaining: 2398, rate_limit_reset: 1462469393}}
```

**Comment by jacegu** [comment]
Get record working:

```
iex(5)> Dnsimple.ZonesService.record(client, 63, "aaa-0001.com", 149267)                                           {:ok,
 %Dnsimple.Response{data: %Dnsimple.Record{content: "mxa.example.com",
   created_at: "2016-05-05T16:33:55.810Z", id: 149267, name: "www",
   parent_id: nil, priority: 10, system_record: false, ttl: 600, type: "MX",
   updated_at: "2016-05-05T16:33:55.810Z", zone_id: "aaa-0001.com"},
  http_response: %HTTPoison.Response{body: "{\"data\":{\"id\":149267,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"www\",\"content\":\"mxa.example.com\",\"ttl\":600,\"priority\":10,\"type\":\"MX\",\"system_record\":false,\"created_at\":\"2016-05-05T16:33:55.810Z\",\"updated_at\":\"2016-05-05T16:33:55.810Z\"}}",
   headers: [{"Server", "nginx"}, {"Date", "Thu, 05 May 2016 16:36:09 GMT"},
    {"Content-Type", "application/json; charset=utf-8"},
    {"Transfer-Encoding", "chunked"}, {"Connection", "keep-alive"},
    {"Status", "200 OK"}, {"X-RateLimit-Limit", "2400"},
    {"X-RateLimit-Remaining", "2397"}, {"X-RateLimit-Reset", "1462469394"},
    {"ETag", "W/\"1cd90f69f4e3dc9afeb5f3aa134c1489\""},
    {"Cache-Control", "max-age=0, private, must-revalidate"},
    {"X-Request-Id", "09afd87e-2231-40e3-b33e-6f1914c95260"},
    {"X-Runtime", "0.197124"},
    {"Strict-Transport-Security", "max-age=31536000"}], status_code: 200},
  rate_limit: 2400, rate_limit_remaining: 2397, rate_limit_reset: 1462469394}}
```

**Comment by weppos** [comment]
It looks good to me now. 👍

**Comment by jacegu** [comment]
Update working:

```
iex(6)> Dnsimple.ZonesService.update_record(client, 63, "aaa-0001.com", 149267, %{ttl: 3600, priority: 20})
{:ok,
 %Dnsimple.Response{data: %Dnsimple.Record{content: "mxa.example.com",
   created_at: "2016-05-05T16:33:55.810Z", id: 149267, name: "www",
   parent_id: nil, priority: 20, system_record: false, ttl: 3600, type: "MX",
   updated_at: "2016-05-05T16:38:44.648Z", zone_id: "aaa-0001.com"},
  http_response: %HTTPoison.Response{body: "{\"data\":{\"id\":149267,\"zone_id\":\"aaa-0001.com\",\"parent_id\":null,\"name\":\"www\",\"content\":\"mxa.example.com\",\"ttl\":3600,\"priority\":20,\"type\":\"MX\",\"system_record\":false,\"created_at\":\"2016-05-05T16:33:55.810Z\",\"updated_at\":\"2016-05-05T16:38:44.648Z\"}}",
   headers: [{"Server", "nginx"}, {"Date", "Thu, 05 May 2016 16:38:44 GMT"},
    {"Content-Type", "application/json; charset=utf-8"},
    {"Transfer-Encoding", "chunked"}, {"Connection", "keep-alive"},
    {"Status", "200 OK"}, {"X-RateLimit-Limit", "2400"},
    {"X-RateLimit-Remaining", "2396"}, {"X-RateLimit-Reset", "1462469393"},
    {"ETag", "W/\"6415adbd30b9e6cb446816617650f0cf\""},
    {"Cache-Control", "max-age=0, private, must-revalidate"},
    {"X-Request-Id", "3eab22b8-c2fe-448d-82f4-858899af2c83"},
    {"X-Runtime", "0.177622"},
    {"Strict-Transport-Security", "max-age=31536000"}], status_code: 200},
  rate_limit: 2400, rate_limit_remaining: 2396, rate_limit_reset: 1462469393}}
```

**Comment by jacegu** [comment]
Delete record working:

```
iex(2)> Dnsimple.ZonesService.delete_record(client, 63, "aaa-0001.com", 149267)
{:ok,
 %Dnsimple.Response{data: nil,
  http_response: %HTTPoison.Response{body: "",
   headers: [{"Server", "nginx"}, {"Date", "Thu, 05 May 2016 16:39:54 GMT"},
    {"Connection", "keep-alive"}, {"Status", "204 No Content"},
    {"X-RateLimit-Limit", "2400"}, {"X-RateLimit-Remaining", "2395"},
    {"X-RateLimit-Reset", "1462469394"}, {"Cache-Control", "no-cache"},
    {"X-Request-Id", "a3a2bfd8-8f5d-4c01-8e55-0604d0c09305"},
    {"X-Runtime", "0.146989"},
    {"Strict-Transport-Security", "max-age=31536000"}], status_code: 204},
  rate_limit: 2400, rate_limit_remaining: 2395, rate_limit_reset: 1462469394}}
```

**Pull request merged**